src/viewport/index.tsx

```diff
@@ -44,8 +44,8 @@ type ViewportProps = {
     showCameraControls: boolean;
     onError?: (error: FrontEndError) => void;
     lockedCamera?: boolean;
-    recording: boolean;
-    onRecordedMovie: (blob: Blob) => void;
+    recording?: boolean;
+    onRecordedMovie?: (blob: Blob) => void; // provide a callback to enable recording
```

Comment:
It might be helpful to have a comment near these two variables to describe how they are intended to interact.  Or maybe at the point where they are being passed in to the component.
The reason I ask is because it's easy to wonder: what if recording is true but onRecordedMovie is undefined? what's stopping that from happening?

Reply:
Right good point. If the feature has not been enabled by providing a callback, then different values for `recording` will have no effect, it serves as an on/off switch for an enabled recorder.

Reply:
I'll add comments, but if its preferred we could change it so that instead of receiving a boolean flag, we hold that value in `ViewportState` and expose `startRecording` and `stopRecording` methods that set that state...